docs: add ai form filler articles (#5628)
vaadin/docs: articles/flow/ai-support/index.adoc

Reviewer: I _still_ feel like _AI Support_ is a weird name for this section. I feel that it can all too easily be mistaken to mean "Getting help from AI with building apps" or "Adding support for AI dev tools" or something.

IMO _AI Integration_ would be a better name.

Author: Updated to use AI Integration

* **Orchestrator** -- [classname]`AIOrchestrator` is a non-visual coordination engine that connects UI components to an LLM provider. It has no DOM element and should not be added to a layout.
* **LLM Provider** -- plugs the orchestrator into any LLM framework. Spring AI and LangChain4j are built in via [classname]`SpringAILLMProvider` and [classname]`LangChain4JLLMProvider`; add others by implementing the [classname]`LLMProvider` interface.
* **Component interfaces** -- [classname]`AIInput`, [classname]`AIMessageList`, [classname]`AIMessage`, and [classname]`AIFileReceiver` define contracts for UI components that the orchestrator can work with. The builder also accepts standard Vaadin components ([classname]`MessageInput`, [classname]`MessageList`, [classname]`UploadManager`, [classname]`Upload`) directly.

> * **Controllers** -- [classname]`AIController` is the framework-agnostic interface for contributing tools and lifecycle hooks to the orchestrator. Built-in controllers bring AI-powered data exploration to [classname]`Grid` and [classname]`Chart` through [classname]`GridAIController` and [classname]`ChartAIController` (backed by a [classname]`DatabaseProvider`), and form-filling to any form layout through [classname]`FormAIController`.

Reviewer: The phrase _"any form layout"_ makes it sound like it only works with FormLayout.

```suggestion
* **Controllers** -- [classname]`AIController` is the framework-agnostic interface for contributing tools and lifecycle hooks to the orchestrator. Built-in controllers bring AI-powered data exploration to [classname]`Grid` and [classname]`Chart` through [classname]`GridAIController` and [classname]`ChartAIController` (backed by a [classname]`DatabaseProvider`), and form-filling to any layout containing input fields through [classname]`FormAIController`.
```

|<<{articles}/components/charts#,Charts>>
|Built and updated from the application database via natural-language requests. See <<ai-powered-chart#,AI-Powered Chart>>.

|<<{articles}/components/form-layout#,Form Layout>>

> |Populated from natural-language input or attached files. See <<ai-powered-form#,AI-Powered Form>>.

Reviewer: ```suggestion
|Fields can be filled in from natural-language input or attached files. See <<ai-powered-form#,AI-Powered Form>>.
```

> |Fields can be filled in from natural-language input or attached files. See <<ai-powered-form#,AI-Powered Form>>.

Reviewer: ```suggestion
|Fields can be filled in from natural-language input or attached files. See <<ai-powered-form#,AI Form Filler>>.
```
